service/client_state.cc

```diff
@@ -224,38 +224,77 @@ future<> service::client_state::has_access(const sstring& ks, auth::command_desc
                 ks + " can be granted only SELECT or DESCRIBE permissions to a non-superuser.");
     }
 
+    if (cmd.resource.kind() == auth::resource_kind::data && cmd.permission == auth::permission::SELECT && is_vector_indexed) {
+
+        co_return co_await ensure_has_permission<auth::command_desc_with_permission_set>({auth::permission_set::of<auth::permission::SELECT, auth::permission::VECTOR_SEARCH_INDEXING>(), cmd.resource});
+
+    }
+
     co_return co_await ensure_has_permission(cmd);
 }
 
-future<bool> service::client_state::check_has_permission(auth::command_desc cmd) const {
+static bool intersects_permissions(const auth::permission_set& permissions, const auth::command_desc_with_permission_set& cmd) {
+    return permissions.intersects(cmd.permission);
+}
+
+static bool intersects_permissions(const auth::permission_set& permissions, const auth::command_desc& cmd) {
```

Comment:
something not right, one intersects_ func calls permissions.intersects and the other permissions.contains ?

Reply:
but the naming is super confusing, no?